roles/setup_efm/tasks/setup_efm.yml

```diff
@@ -23,6 +23,26 @@
 - name: Reference variables
   include_vars: "{{ pg_type }}_{{ ansible_os_family }}.yml"
 
+- name: Find the WAL dir source location
+  stat:
+    path: "{{ pg_data }}/pg_wal"
+  register: waldir
+
+- name: Find the source of WAL dir if it's symlink
+  shell: >
+    dirname "{{ waldir.stat.lnk_source }}"
+  when:
+    - waldir is defined
+    - waldir.stat.islnk
+  register: wal_src_location
```

Comment:
I think you could extract parent folder path with the help of the `dirname` jinja filter:
```yaml
- name: Find the source of WAL dir if it's symlink
  set_fact:
     wal_src_location: "{{ waldir.stat.lnk_source | dirname }}"
  when:
    - waldir is defined
    - waldir.stat.islnk
```

Reply:
Interesting. Thank you for sharing this.

Reply:
Fixed